### PR TITLE
Proposed new error value for implementations that restrict parser ali…

### PR DESCRIPTION
**Pull request by jafingerhut**
…gnment

Similar to the existing statement that some implemtations are allowed
to restrict headers to be multiples of 8 bits long, such
implementations are also likely to impose similar restrictions on the
2-argument extract and advance calls, so that the parsing 'cursor' is
always at a multiple of 8 bits from the beginning of the packet.
These changes propose a way to make that explicit in the
specification.

### CODE REVIEW

**Comment by jafingerhut** [comment]
Proposed change motivated by implementation work in p4c and BMv2 on advance: https://github.com/p4lang/behavioral-model/issues/727

File: p4-16/spec/P4-16-spec.mdk [hunk 1]
```diff
@@ -5021,6 +5021,8 @@ is captured by the following pseudo-code:
 ~ Begin P4Pseudo
 void packet_in.extract<T>(out T headerLvalue,
                           in bit<32> variableFieldSize) {
+   // implementations _may_ include the following line, but need not
+   verify(variableFieldSize[2:0] == 0, error.ParserInvalidAlignment);
```
Comment (mbudiu-vmw) [inline p4-16/spec/P4-16-spec.mdk]: I think I would prefer this check to be commented-out.

File: p4-16/spec/P4-16-spec.mdk [hunk 2]
```diff
@@ -7090,7 +7094,9 @@ error {
     NoMatch,           /// 'select' expression has no matches.
     StackOutOfBounds,  /// Reference to invalid element of a header stack.
     HeaderTooShort,    /// Extracting too many bits into a varbit field.
-    ParserTimeout      /// Parser execution time limit exceeded.
+    ParserTimeout,     /// Parser execution time limit exceeded.
+    ParserInvalidAlignment  /// extract or advance would leave parser with
```
Comment (mbudiu-vmw) [inline p4-16/spec/P4-16-spec.mdk]: This error name is very specific. How about InvalidArgument, which could probably be reused for other purposes?

**Comment by jafingerhut** [comment]
Agreed with all you said in the previous two comments.  No rush on my part to get it merged.  Just wanted to create the PR while the issue was fresh in mind.

File: p4-16/spec/P4-16-spec.mdk (hunk 1 above)
Comment (jafingerhut) [reply p4-16/spec/P4-16-spec.mdk]: done

File: p4-16/spec/P4-16-spec.mdk (hunk 2 above)
Comment (jafingerhut) [reply p4-16/spec/P4-16-spec.mdk]: done

**Comment by jafingerhut** [comment]
AI Andy: Add this change to the change log at the end of the spec.  Replace "Implementations" with "Targets".

**Comment by jafingerhut** [comment]
I believe I was verbally given the go-ahead to merge this in after addressing the comments made during the 2019-Apr-01 LDWG meeting.  If I am mistaken, or further changes or needed, let me know and I can fix things.

**Comment by jnfoster** [comment]
👍

**Pull request merged**